app/controllers/advertisements_controller.rb

```diff
@@ -0,0 +1,28 @@
+class AdvertisementsController < ApplicationController
+  def index
+    @advertisement = Advertisement.all
+  end
+
+  def show
+    @advertisement = Advertisement.find(params[:id])
+  end
+
+  def new
+    @advertisement = Advertisement.new
+  end
+
+  def create
+    @advertisement = Advertisement.new
```

Comment:
OK, I see why you need it, but that feels wrong. IMO, create should never render, only redirect. Can you ask your mentor about that?

What I mean is that if the create failed, it should redirect back to the new action with the error message (and the args so you can fill out the form again). But rendering a view from the create endpoint feels very wrong.